src/main/java/org/dita/dost/reader/CopyToReader.java

```diff
@@ -86,11 +87,12 @@ public void startElement(final String uri, final String localName, final String
         processRoleStack.push(processingRole);
 
         final String classValue = atts.getValue(ATTRIBUTE_NAME_CLASS);
-
-        if (foreignLevel > 0) {
-            foreignLevel++;
-        } else if (TOPIC_FOREIGN.matches(classValue) || TOPIC_UNKNOWN.matches(classValue)) {
-            foreignLevel++;
+        
+        final DitaClass cls = atts.getValue(ATTRIBUTE_NAME_CLASS) != null ? new DitaClass(atts.getValue(ATTRIBUTE_NAME_CLASS)) : new DitaClass("");
+        if (cls.isValid()) {
+        	classes.addFirst(cls);
```

Comment:
`addFirst` adds to the head of the Deque, whereas the  `add` in the else branch adds to the tails of the Deque and is equivalent to `addLast`. These should both be `addFirst`